dlup/cli/mask.py

```diff
@@ -0,0 +1,197 @@
+        if curr_mask.sum() == 0:
+            continue
+        output[index_map[index]] = mask_to_polygons(_mask, offset=sample["coordinates"], scaling=scaling)
+    return output
+
+
+def dataset_to_polygon(dataset, index_map, num_workers=0, scaling=1.0, show_progress=True):
+    output_polygons: dict[str, list[Polygon]] = {v: [] for v in index_map.values()}
+
+    sample_function = partial(_get_sample, dataset=dataset, index_map=index_map, scaling=scaling)
+
+    if num_workers <= 0:
+        for idx in tqdm(range(len(dataset)), disable=not show_progress):
+            curr_polygons = sample_function(idx)
+            for polygon_name in output_polygons:
+                if polygon_name in curr_polygons:
+                    output_polygons[polygon_name] += curr_polygons[polygon_name]
+    else:
+        with Pool(num_workers) as pool:
+            with tqdm(total=len(dataset), disable=not show_progress) as pbar:
+                for curr_polygons in pool.imap(sample_function, range(len(dataset))):
+                    pbar.update()
+                    for polygon_name in output_polygons:
+                        if polygon_name in curr_polygons:
+                            output_polygons[polygon_name] += curr_polygons[polygon_name]
+
+    geometry = {k: unary_union(polygons) for k, polygons in output_polygons.items()}
+    return geometry
+
+
+def mask_to_polygon(args: argparse.Namespace):
+    """Perform the mask conversion to polygon."""
+    mask_filename = args.MASK_FN
+    output_filename = args.OUTPUT_FN
+    tile_size = cast(tuple[int, int], (args.tile_size,) * 2)
+    tile_overlap = cast(tuple[int, int], (args.tile_overlap,) * 2)
+
+    # Prepare output directory.
+    output_filename.parent.mkdir(parents=True, exist_ok=True)
+
+    dataset = TiledROIsSlideImageDataset.from_standard_tiling(
+        mask_filename,
+        tile_size=tile_size,
+        tile_overlap=tile_overlap,
+        mpp=None,
+        tile_mode=TilingMode.overflow,
+        crop=False,
+        backend=ImageBackend.TIFFFILE,
+        interpolator=Resampling.NEAREST,
+    )
+    target_mpp = args.mpp
+    if target_mpp is None:
+        scaling = 1.0
+    else:
+        scaling = dataset.slide_image.get_scaling(target_mpp=target_mpp)
+
+    # Parse the labels
+    if args.labels is None:
+        index_map = {1: "label"}
+    else:
+        index_map = {}
+        for pair in args.labels.split(","):
+            name, index = pair.split("=")
+            if not index.isnumeric():
+                raise argparse.ArgumentTypeError(f"Expected a key-pair of the form 1=tumor,2=stroma")
+            index = float(index)
+            if not index.is_integer():
+                raise argparse.ArgumentTypeError(f"Expected a key-pair of the form 1=tumor,2=stroma")
+            index = int(index)
+            if index == 0:
+                raise argparse.ArgumentTypeError(f"0 is not a proper index. Needs to be at least 1.")
+            index_map[index] = name.strip()
```

Comment:
can we move this to a generic function in `utils.cli`? seems not related to the actual functionality of wsi-to-polygon, but just an argparse thing.